qiskit/circuit/parameterexpression.py

```diff
@@ -508,14 +508,21 @@ def __eq__(self, other):
     def is_real(self):
         """Return whether the expression is real"""
 
-        if not self._symbol_expr.is_real and self._symbol_expr.is_real is not None:
+        # workaround for symengine behavior that const * (0 + 1 * I) is not real
+        # see https://github.com/symengine/symengine.py/issues/414
+        if _optionals.HAS_SYMENGINE and self._symbol_expr.is_real is None:
+            symbol_expr = self._symbol_expr.evalf()
+        else:
+            symbol_expr = self._symbol_expr
+
+        if not symbol_expr.is_real and symbol_expr.is_real is not None:
```

Comment:
https://github.com/symengine/symengine/issues/1922 seems fixed now. Is the workaround still needed?

Reply:
@1ucian0 This commit is actually not related to this PR. It was simply pulled in from a `git merge main` just after the PR was initially opened. The changes relevant to this PR are to  [`_state_to_latex_ket()`](https://github.com/ryanhill1/qiskit-terra/blob/7fc0ce52e662d9605eaf265cdd25159c0e443340/qiskit/visualization/state_visualization.py#L1273) and [`test_state_to_latex_with_prefix()`](https://github.com/ryanhill1/qiskit-terra/blob/7fc0ce52e662d9605eaf265cdd25159c0e443340/test/python/quantum_info/states/test_statevector.py#L1133). As of version [`0.22.0.dev0+6927a8d`](https://github.com/Qiskit/qiskit-terra/tree/6927a8d26bb58bf402b4e38f5563ed97a76a0fad) (5 Sept. 2022) the error from issue #8460 is still present, but would be resolved by this PR.

Reply:
I think your merge might not have gone quite right - it looks like you've cherry-picked the last commit rather than merging the whole branch.  You might want to try rebasing your branch over our `main` instead.

Reply:
@jakelishman Thanks for the tip! Should be fixed now